fix(Chat): remove hardcoded prices and quantity

nwjbrandon opened the pull request:
This patches address Issue #22 by removing the hardcoded prices
and quantity. These prices and quantity are obtained from the
socket response. The changes made are to the chat rooms and chat
header

nwjbrandon committed:
fix(Price and Quantity): remove hardcoded prices and quantity

This patches address Issue #22 by removing the hardcoded prices
and quantity. These prices and quantity are obtained from the
socket response. The changes made are to the chat rooms and chat
header

karrui commented on src/routes/chat/ChatHeader/ChatHeader.jsx:
```diff
@@ -87,19 +93,50 @@ const ChatCreateOffer = ({ setOffer }) => {
 
 const ChatHeader = () => {
   const [offer, setOffer] = useState(true);
```
Didn't catch this the last time, but for booleans please add a `is` prefix; i.e `isOffer, setIsOffer`

karrui commented on src/routes/chat/ChatHeader/ChatHeader.jsx:
```diff
@@ -87,19 +93,50 @@ const ChatCreateOffer = ({ setOffer }) => {
+    buyerNumberOfShares
+  } = useSelector(state => state.chat.chatConversation);
+  const userType = useSelector(state => state.misc.userType);
+  const otherPartyUserType = getOtherPartyUserType({ userType });
```
Suggest not using objects as parameters for normal functions, just `getOtherPartyUserType(userType)` will do.

karrui commented on src/routes/chat/ChatHeader/ChatHeader.jsx:
```diff
@@ -87,19 +93,50 @@ const ChatCreateOffer = ({ setOffer }) => {
+  } = useSelector(state => state.chat.chatConversation);
+  const userType = useSelector(state => state.misc.userType);
+  const otherPartyUserType = getOtherPartyUserType({ userType });
+  const userPrice = getUserPrice({
```
Same for this one and all the other functions, just 3 parameters will do, without transforming it to an object

karrui commented on src/routes/chat/ChatHeader/ChatHeader.jsx:
```diff
@@ -87,19 +93,50 @@ const ChatCreateOffer = ({ setOffer }) => {
-          headerText="Seller Offer"
-          quantity="2000"
-          price="6.89"
+          headerText={`${userType === 'seller' ? 'Buyer' : 'Seller'} Offer`}
```
We have constants for user types, `BUYER` and `SELLER`. Check `constants/user.js`

karrui commented on src/routes/chat/ChatHeader/ChatHeader.jsx:
```diff
@@ -87,19 +93,50 @@ const ChatCreateOffer = ({ setOffer }) => {
-          quantity="2000"
-          price="6.89"
+          headerText={`${userType === 'seller' ? 'Buyer' : 'Seller'} Offer`}
+          quantity={toSgdCurrency(otherPartyNumberOfShares)}
```
Don't use this function here, number of shares are not currencies.

karrui commented on src/routes/chat/ChatHeader/ChatHeader.jsx:
```diff
@@ -87,19 +93,50 @@ const ChatCreateOffer = ({ setOffer }) => {
+        />
+        <ChatOfferDetails
+          headerText="Your Bid"
+          quantity={toSgdCurrency(userNumberOfShares)}
```
Same here.

karrui commented on src/routes/chat/ChatRooms/ChatRoom.jsx:
```diff
@@ -48,15 +55,43 @@ const ChatRoom = ({ chat, basePath }) => {
+    buyerNumberOfShares
+  } = chat;
+  const userType = useSelector(state => state.misc.userType);
+  const otherPartyUserType = getOtherPartyUserType({ userType });
```
Same here, pass in the parameter by itself rather than an object.

karrui commented on src/routes/chat/ChatRooms/ChatRoom.jsx:
```diff
@@ -48,15 +55,43 @@ const ChatRoom = ({ chat, basePath }) => {
+  return (
+    <div className="detail__content">
+      <div>
+        {userType === 'seller' ? 'Selling' : 'Buying'} Amt: {numberOfShares}
```
Use constants

karrui commented on src/utils/userUtils.js:
```diff
@@ -18,10 +18,29 @@ export const isCommittee = user => {
   return user.isCommittee;
 };
 
+export const getUserPrice = ({ userType, sellerPrice, buyerPrice }) => {
```
Accept a parameter straight, 

```
export const getUserPrice = (userType, sellerPrice, buyerPrice) => {
```

karrui commented on src/utils/userUtils.js:
```diff
@@ -18,10 +18,29 @@ export const isCommittee = user => {
 };
 
+export const getUserPrice = ({ userType, sellerPrice, buyerPrice }) => {
+  return userType === 'seller' ? sellerPrice : buyerPrice;
```
Use constant `SELLER`

karrui commented on src/utils/userUtils.js:
```diff
@@ -18,10 +18,29 @@ export const isCommittee = user => {
+  return userType === 'seller' ? sellerPrice : buyerPrice;
+};
+
+export const getUserNumberOfShares = ({
```
Same here

karrui commented on src/utils/userUtils.js:
```diff
@@ -18,10 +18,29 @@ export const isCommittee = user => {
+  return userType === 'seller' ? sellerNumberOfShares : buyerNumberOfShares;
+};
+
+export const getOtherPartyUserType = ({ userType }) => {
```
`export const getOtherPartyUserType = (otherPartyType) => {`

nwjbrandon committed:
Merge branch 'staging' into fix-hardcoded-price-and-quantity

nwjbrandon committed:
fix(Accept Offer)

This patch uses the user constants, change the function to take in
normal arguments instead of objects, and fixes the quantity which
was previously displayed as currency.

nwjbrandon committed:
fix(Accept Offer): use constants SELLER and BUYER in userUtils

This commit uses SELLER and BUYER implemented in the user constants
in userUtils

karrui commented on src/routes/chat/ChatHeader/ChatHeader.jsx:
```diff
@@ -86,23 +93,50 @@ const ChatCreateOffer = ({ setOffer }) => {
+  } = useSelector(state => state.chat.chatConversation);
+  const userType = useSelector(state => state.misc.userType);
+  const otherPartyUserType = getOtherPartyUserType(userType);
+  const userPrice = getUserPrice(otherPartyUserType, sellerPrice, buyerPrice);
```
Is this intended? userPrice but using the otherPartyUserType?

nwjbrandon committed:
fix(Accept Offer): fix wrong userType

Fixes wrong userType pass into the function to get the price and quantity
in chatheader.jsx

nwjbrandon committed:
fix(Offer): fix bug in assigning the other party user type
